src/nimib/sources.nim

```diff
@@ -6,29 +6,36 @@ import std/[
 import types
 
 
-
 # Credits to @haxscramper for sharing his code on reading the line info
 # And credits to @Yardanico for making a previous attempt which @hugogranstrom have taken much inspiration from
 # when implementing this.
 
 type
   Pos* = object
+    filename*: string
     line*: int
     column*: int
 
+proc `<`*(p1, p2: Pos): bool =
+  (p1.line, p1.column) < (p2.line, p2.column)
```

Comment:
I am guessing since you checked on quite a few examples it is not something that happens often or possibly ever, but I gotta ask. is it possible we end comparing positions from two different files? should we check that this does not happen and put an assert? and in case it happens? I guess we should only compare stuff that comes from same file.

Reply:
It is totally possible that we compare from two different files. I'll add an assert with a prompt to open an issue if it ever happens. :+1: